### PR TITLE
popup component

### PR DESCRIPTION
**Pull request by sonac**

### CODE REVIEW

File: app/assets/javascripts/src/PicCompare.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      showedPairs: [[]],
+      picRtngs: {}
+    };
+    this.getNextPair = this.getNextPair.bind(this)
+  };
+
+  componentDidMount() {
+    fetch('/api/pictures')
+      .then(response => response.json())
+      .then(data => this.setState({ picUrls: data }));
```
Comment (citizen4371) [inline app/assets/javascripts/src/PicCompare.jsx:18]: .then(picUrls => this.setState(state => ({...state, picUrls})))

File: app/assets/javascripts/src/PicCompare.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  getNextPair() {
+    const newPair = this.state.curPics.map(x => {
+      return x + 2
+    })
+    this.setState({curPics: newPair})
```
Comment (citizen4371) [inline app/assets/javascripts/src/PicCompare.jsx:25]: this.setState(state => ({...state, curPics: state.curPics.map(x => x + 2)}))

Comment (citizen4371) [reply app/assets/javascripts/src/PicCompare.jsx:25]: вместо всего тела функции

File: app/assets/javascripts/src/PicCompare.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  render() {
```
Comment (citizen4371) [inline app/assets/javascripts/src/PicCompare.jsx:28]: const { picUrls, curPics } = this.state;

File: app/assets/javascripts/src/PicCompare.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    return (
+      <div>
+        <img src={this.state.picUrlsp[curPics[0]]} alt="First pic"> <img src={this.state.picUrlsp[curPics[1]]} alt="Second pic">
```
Comment (citizen4371) [inline app/assets/javascripts/src/PicCompare.jsx:31]: this.state.picUrlsp[curPics[0]] поменяй на picUrls[curPics[0]]

**Review by citizen4371 — CHANGES_REQUESTED** [review]

**Comment by eillasti** [comment]
outdated?

**Comment by sonac** [comment]
yep

**Pull request closed without merging**